pkg/operator/ceph/cluster/watcher.go

```diff
@@ -130,6 +170,122 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
 	return false
 }
 
+func (c *clientCluster) createNetworkFence(ctx context.Context, node corev1.Node, clusterName string) error {
+	nodeIPtoBlock, pv, err := c.getWatcherIpPortToBlock(ctx, node, clusterName)
+	if err != nil {
+		return fmt.Errorf("failed to get node IP to block for node %q. %v", node.Name, err)
+	}
+	fmt.Println("NodeIP to block", nodeIPtoBlock)
+
+	networkFence := &addonsv1alpha1.NetworkFence{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      node.Name,
+			Namespace: c.namespace,
+		},
+		Spec: addonsv1alpha1.NetworkFenceSpec{
+			Driver:     "rook-ceph.rbd.csi.ceph.com",
+			FenceState: addonsv1alpha1.Fenced,
+			Secret: addonsv1alpha1.SecretSpec{
+				Name:      pv.Spec.CSI.ControllerPublishSecretRef.Name,
+				Namespace: pv.Spec.CSI.ControllerPublishSecretRef.Namespace,
+			},
+			Cidrs: []string{
+				nodeIPtoBlock,
+			},
+			Parameters: map[string]string{
+				"clusterID": pv.Spec.CSI.VolumeAttributes["clusterID"],
```

Comment:
If this CR is created in the cephcluster namespace, how are you ensuring you are using the clusterID belongs to this ceph cluster not to the other ceph cluster? in other words how are you ensuring the PV is created for this ceph cluster not for the other cluster?

Reply:
since we are creating by fetching the driver based on the  pv namespace and same pv `clusterID` is used it should be fine right?